hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/MorphingPortableReader.java

```diff
@@ -198,10 +198,22 @@ public short readShort(String fieldName) throws IOException {
     }
 
     @Override
+    public boolean[] readBooleanArray(String fieldName) throws IOException {
+        FieldDefinition fd = cd.getField(fieldName);
+        if (fd == null) {
+            return null;
+        }
+        if (fd.getType() != FieldType.BOOLEAN_ARRAY) {
+            throw new IncompatibleClassChangeError();
```

Comment:
I don't think we put each method doc that it may return null. Our serialization has a contract of "what you put , that you get" . That I will put into tutorial.